NuGet.Services.Dashboard.Operations/Infrastructure/SecretConfigurationManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using System.Configuration;
+using NuGet.Services.KeyVault;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace NuGet
+{
+    public static class SecretConfigurationManager
+    {
+        private static ISecretInjector _secretInjector;
+        private static ISecretReader _keyVaultReader;
+
+        public static void Init()
+        {
+            var vaultName = ConfigurationManager.AppSettings["VaultName"];
+            var clientId = ConfigurationManager.AppSettings["ClientId"];
+            var certificateThumbprint = ConfigurationManager.AppSettings["CertificateThumbprint"];
+            var validateCertificate = ConfigurationManager.AppSettings["ValidateCertificate"].Equals("true", StringComparison.InvariantCultureIgnoreCase) ? true : false;
+
+            _keyVaultReader = new KeyVaultReader(new KeyVaultConfiguration(vaultName, clientId, certificateThumbprint, validateCertificate));
+            _secretInjector = new SecretInjector(_keyVaultReader);
+
+            IEnumerable keys = new List<string>(ConfigurationManager.AppSettings.AllKeys);
+
+            foreach (string key in keys)
+            {
+                string newValue;
+                if (TryGetConfigurationValue(key, out newValue))
```

Comment:
You could add this to `TryGetConfigurationValue` rather than having to deal with passing the config value back here, e.g.

```
try
{
    result = _secretInjector.InjectAsync(settingValue).Result;
    ConfigurationManager.AppSettings.Set(key, newValue);
}
```
Might want to rename the method then as well, since it would no longer follow our "TryGet" idiom.